### PR TITLE
Embeddings feature

### PR DESCRIPTION
**Pull request by inayrus**
Hiya, would this work for the Bert sentence embeddings? What do you think would be a good number to use for the max tokens per sequence?

### CODE REVIEW

File: feature_embeddings.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        self.tag2idx = self.create_tag_dict()
+        self.max_len = max_sent_len
+
+    def load_data(self, path):
+        data = pd.read_csv(path,
+                           delimiter="\t",
+                           names=["corpus", "n_sent", "n_word", "token", "tag"])
+        return data
+
+    def create_tag_dict(self):
```
Comment (lagewel001) [inline feature_embeddings.py:55]: Kind off overkill for 4 predetermined tags, but nice nonetheless ;)

File: feature_embeddings.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        # convert the tags to indices and create dict for conversion
+        tag_values = list(set(self.data["tag"].values))
+        tag_values.append("PAD")
+        tag2idx = {t: i for i, t in enumerate(tag_values)}
+        return tag2idx
+
+    def tokenize_and_preserve_labels(self, sentence, text_labels):
```
Comment (lagewel001) [inline feature_embeddings.py:62]: Nice

File: feature_embeddings.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+import torch
+
+class SentenceGetter(object):
+
+    def __init__(self, data):
+        self.n_sent = 1
+        self.data = data
+        self.empty = False
+        # todo: add other lexical features (w, t, pos, lemma, stem)
+        agg_func = lambda s: [(w, t) for w, t in zip(s["token"].values.tolist(),
```
Comment (lagewel001) [inline feature_embeddings.py:26]: This part seems needlessly complicated in combination with the multiple times labels and their tags are zipped and unzipped again (starting from line 87), but if it ain't broke...

**Review by lagewel001 — APPROVED** [review]
Looks nice and neat. I think this can be a good setup for further steps. I'll look into the training with the extra features. Thanks!!

File: feature_embeddings.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            ids = [101] + ids + [102]
+            target_tag = [self.tag2idx["PAD"]] + target_tag + [self.tag2idx["PAD"]]
+
+            # pad shortened sequences
+            padding_len = self.max_len - len(ids)
+            all_ids[i] = ids + ([0] * padding_len)
+            all_target_tag[i] = target_tag + ([self.tag2idx["PAD"]] * padding_len)
+
+        # creating the attention mask
+        attention_mask = [[float(i != 0.0) for i in ii] for ii in all_ids]
```
Comment (boczekbartek) [inline feature_embeddings.py:123]: This is a bit dangerous, you should never compare floats like this, but here is float(int != float)?? A
 bit sketchy line which may bring unexpected output

**Review by boczekbartek — APPROVED** [review]
Very nice! Good base for the future :)
